fatjdbc/src/test/java/com/qubole/quark/fatjdbc/test/MaterializedViewJoinTest.java

```diff
@@ -0,0 +1,183 @@
+package com.qubole.quark.fatjdbc.test;
+
+import com.qubole.quark.catalog.db.encryption.MysqlAES;
+import com.qubole.quark.planner.parser.SqlQueryParser;
+import com.qubole.quark.sql.ResultProcessor;
+import org.apache.calcite.sql.SqlDialect;
+import org.flywaydb.core.Flyway;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by amoghm on 3/21/16.
+ */
+public class MaterializedViewJoinTest {
+  private static Connection h2Connection;
+  protected static String dbUrl = "jdbc:h2:mem:MaterializedViewJoinTest;DB_CLOSE_DELAY=-1";
+
+  private static final String dbSchemaUrl = "jdbc:h2:mem:MaterializedViewJoinTest1;DB_CLOSE_DELAY=-1";
+  private static Connection dbConnection;
+  private static Properties connInfo;
+
+  @BeforeClass
+  public static void setUpDb() throws Exception {
+    MaterializedViewJoinTest.setUpClass(dbUrl);
+    Flyway flyway = new Flyway();
+    flyway.setDataSource(dbSchemaUrl, "sa", "");
+    flyway.migrate();
+
+    // Encrypting url, username and password before storing in db
+    MysqlAES mysqlAES = MysqlAES.getInstance();
+    mysqlAES.setKey("key");
+    String url = mysqlAES.convertToDatabaseColumn(dbUrl);
+    String username = mysqlAES.convertToDatabaseColumn("sa");
+    String password = mysqlAES.convertToDatabaseColumn("");
+
+    connInfo = new Properties();
+    connInfo.setProperty("url", dbSchemaUrl);
+    connInfo.setProperty("user", "sa");
+    connInfo.setProperty("password", "");
+    connInfo.setProperty("db:" +
+        MaterializedViewJoinTest.class.
+            getResource("/HiveUDFModel.json").getPath(), "");
+    connInfo.setProperty("schemaFactory",
+        "com.qubole.quark.catalog.db.SchemaFactory");
```

Comment:
FYI: In tests its OK to use json schema factory. Seems like its easier to setup than db.